app/src/main.c

```diff
@@ -1,8 +1,10 @@
-#include <zephyr/device.h>
-#include <zephyr/devicetree.h>
-#include <zephyr/drivers/gpio.h>
-#include <zephyr/logging/log.h>
-#include <zephyr/zephyr.h>
+//! @file
```

Comment:
rip :coffin:

Reply:
guess we should add the compatibility hack to this repo so this can build on 3.1?

Reply:
i vote adding this sad patch to CMakeLists.txt as well to support 3.2:
```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index e2a8286..0166210 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,15 @@
-# Copyright (c) 2021 Nordic Semiconductor ASA
-# SPDX-License-Identifier: Apache-2.0
-#
-# This CMake file is picked by the Zephyr build system because it is defined
-# as the module CMake entry point (see zephyr/module.yml).
+# This CMake file is picked by the Zephyr build system because it is defined as
+# the module CMake entry point (see zephyr/module.yml).
+
+# The Zephyr includes moved from /include to /include/zephyr in Zephyr 3.2:
+# https://github.com/zephyrproject-rtos/zephyr/commit/53ef68d4598b2f9005c5da3fc0b860ca1999d350
+# Add the old path for backwards compatibility. Note that Zephyr itself supports
+# the Kconfig option 'CONFIG_LEGACY_INCLUDE_PATH' to enable the same
+# compatibility behavior, but it's marked as deprecated and to be removed in the
+# future, so just apply the compatibiltiy fix here.
+if(${KERNEL_VERSION_MAJOR}.${KERNEL_VERSION_MINOR}.${KERNEL_PATCHLEVEL}
+   VERSION_GREATER_EQUAL
+   "3.1.99"
+)
+  zephyr_include_directories(${ZEPHYR_BASE}/include/zephyr)
+endif()
```

Reply:
fyi @chrisc11 i shoved in that patch to the `memfault` branch today since i needed to build this against zephyr 2.7.3 :+1: